tools/harness-simulation/harness/Thread_Harness/Sniffer/SimSniffer.py

```diff
@@ -66,12 +152,40 @@ def __init__(self, **kwargs):
         self._local_pcapng_location = None
 
         if self.addr_port is not None:
+            # Replace `MsgBox` only once when simulation devices exist
+            if not SimSniffer.replaced_msgbox:
+                SimSniffer.replaced_msgbox = True
+                self._replaceMsgBox()
```

Comment:
Putting this `run once` logic in `_replaceMsgBox` may make the constructor neater.